fix: fix: resolve WebView ID–based account-change issues on Native

huhuanming opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Performance Improvements**
  * Staggered tab mounting introduced; lazy loading disabled for more consistent startup
  * Removed background tab preloading to reduce resource use

* **Improvements**
  * DApp notifications now support tab-aware contexts
  * Views defer mounting until focused to reduce unnecessary rendering
  * Desktop focus defaults adjusted for more predictable drag behavior

* **Behavior Change**
  * Changing Perp user type can trigger navigation between Perp views
  * Closing deposit/withdraw behaves differently on native platforms
  * Notification navigation now includes topic type and emits a log entry
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

huhuanming committed:
Fix Cardano provider import to use default export

Corrects the dynamic import of ProviderApiCardano to access its default export, ensuring the provider is instantiated properly.

huhuanming committed:
Update Navigator.tsx

huhuanming committed:
Update TabStackNavigator.tsx

huhuanming committed:
Revert "fix: fix permission & skipWebDevicePrompt (#8808)"

This reverts commit a1ce599af7f885a141f98825c4fee0eea0f6da6a.

huhuanming committed:
Reapply "fix: fix permission & skipWebDevicePrompt (#8808)"

This reverts commit 8ad69f2f541ca8d5748d2eb57f6a3835de8f914d.

huhuanming committed:
Update useDAppNotifyChanges.ts

huhuanming committed:
Revert "Update useDAppNotifyChanges.ts"

This reverts commit 9a1ebfcd254e64722afb5c9d9bcc8308c1042fb0.

huhuanming committed:
Update useDAppNotifyChanges.ts

huhuanming committed:
Update useDAppNotifyChanges.ts

huhuanming committed:
Update PageWebviewPerpTrade.tsx

huhuanming committed:
Update PageWebviewPerpTrade.tsx

huhuanming committed:
Update PageWebviewPerpTrade.tsx

huhuanming committed:
Update Perp.tsx

huhuanming committed:
Merge branch 'x' into fix-ProviderApiCardano

revan-zhang commented:
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scanner | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-critical.svg) Critical | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg) High | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-medium.svg) Medium | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-low.svg) Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/3a9f945e-d6f4-429d-a796-5ff18e7e170d?refs=310f93182b420c647f99be2b0d63bfc2f1d594be&source=prChecksComment) |
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/3a9f945e-d6f4-429d-a796-5ff18e7e170d/license?refs=310f93182b420c647f99be2b0d63bfc2f1d594be&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

originalix commented on packages/kit/src/views/Discovery/hooks/useDAppNotifyChanges.ts:
```diff
@@ -159,12 +160,16 @@ export function useDAppNotifyChanges({ tabId }: { tabId: string | null }) {
   }, [tab?.url, isFocusedInDiscoveryTab, previousUrl]);
 
-  const getWebviewRef = useCallback(() => webviewRef, [webviewRef]);
+  const getWebviewRef = useCallback(
+    () => getWebviewWrapperRef(tabId ?? ''),
+    [tabId],
+  );
```
这里为什么要这样写，理论上两种写法是等价的吧

huhuanming replied:
不等价啊，一定概率出现时间差，webviewRef 赋值为null，再跳用getWebviewRef函数，结果就拿不到了

huhuanming committed:
Update index.tsx

huhuanming committed:
Update index.tsx

huhuanming committed:
Merge branch 'fix-ProviderApiCardano' of github.com:OneKeyHQ/app-monorepo into fix-ProviderApiCardano

huhuanming committed:
Update TabStackNavigator.tsx

huhuanming committed:
Update HeaderView.tsx

huhuanming committed:
Update DepositWithdrawModal.tsx

huhuanming committed:
Log notification item clicks with topic type

Added logging for notification item clicks, including the topic type, by updating the notification navigation utility and logger. Also updated type definitions to use ENotificationPushTopicTypes for topicType.

huhuanming committed:
Update Perp.tsx

huhuanming committed:
Fix useEffect dependencies for isMounted logic

Updated useEffect dependencies in PerpView and PageWebviewPerpTradeView to remove isMounted from the dependency array. This prevents unnecessary effect executions and ensures correct mounting behavior.

huhuanming committed:
Update notificationsUtils.ts
